src/v/redpanda/application.cc

```diff
@@ -334,6 +336,15 @@ void application::initialize(
           config::shard_local_cfg().zstd_decompress_workspace_bytes());
     }).get0();
 
+    /*
+     * allocate per-core workspaces for async_stream_zstd. it can be several
+     * megabytes in size, so do it before memory becomes fragmented.
+     */
+    ss::smp::invoke_on_all([] {
+        compression::async_stream_zstd::init_workspace(
+          config::shard_local_cfg().zstd_decompress_workspace_bytes());
```

Comment:
Please add a comment here to note that the double-allocation of workspaces should go away when the non-async compression path is removed: important that we don't forget to do that.

Reply:
Added a `TODO` to remove the double allocation after the other compression path in `v/kafka` is removed.